CIScripts/Tests/TCPCommunicationTest.ps1

```diff
@@ -1,3 +1,67 @@
 function Test-TCPCommunication {
-    # TODO: JW-838: Implement
+    Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session,
+           [Parameter(Mandatory = $true)] [TestConfiguration] $TestConfiguration)
+
+    Write-Host "Running TCP Communication test"
+
+    Initialize-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+
+    $NetworkName = $TestConfiguration.DockerDriverConfiguration.NetworkConfiguration.NetworkName
+
+    Write-Host "Creating containers"
+    $ServerID, $ClientID = Invoke-Command -Session $Session -ScriptBlock {
+        $ServerID = docker run --network $Using:NetworkName -d iis-tcptest
+        $ClientID = docker run --network $Using:NetworkName -d microsoft/nanoserver ping -t localhost
+        return $ServerID, $ClientID
+    }
+
+    . $PSScriptRoot\CommonTestCode.ps1
+
+    Write-Host "Getting VM NetAdapter Information"
+    $VMNetInfo = Get-RemoteNetAdapterInformation -Session $Session -AdapterName $TestConfiguration.AdapterName
+
+    Write-Host "Getting Containers NetAdapter Information"
+    $ServerNetInfo = Get-RemoteContainerNetAdapterInformation -Session $Session -ContainerID $ServerID
+    $ClientNetInfo = Get-RemoteContainerNetAdapterInformation -Session $Session -ContainerID $ClientID
+
+    Write-Host "Configuring..."
+    Invoke-Command -Session $Session -ScriptBlock {
+        vif.exe --add $Using:VMNetInfo.IfName --mac $Using:VMNetInfo.MACAddress --vrf 0 --type physical
+        vif.exe --add HNSTransparent --mac $Using:VMNetInfo.MACAddress --vrf 0 --type vhost --xconnect $Using:VMNetInfo.IfName
+
+        vif.exe --add $Using:ServerNetInfo.AdapterShortName --mac $Using:ServerNetInfo.MACAddress --vrf 1 --type virtual --vif 1
```

Comment:
I'd prefer if you could use higher vif numbers for now, because if for example physical interface has ID 1 in operating systems, this will fail. While it hasn't happened yet (physical host usually has an id between 10 and 20), it wouldn't hurt to specify some out-of-the-world value, like 1000.

This will stop being an issue after the new naming convention is merged, so you can ignore this comment.